python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
         return SparseMatrix(numRows, numCols, colPtrs, rowIndices, values)
 
 
+class DistributedMatrix(object):
+    """
+    Represents a distributively stored matrix backed by one or
+    more RDDs.
+
+    """
+    def numRows(self):
+        """Get or compute the number of rows."""
+        raise NotImplementedError
+
+    def numCols(self):
+        """Get or compute the number of cols."""
+        raise NotImplementedError
+
+
+class RowMatrix(DistributedMatrix):
+    """
+    Represents a row-oriented distributed Matrix with no meaningful
+    row indices.
+
+    .. note:: Experimental
+    """
+    def __init__(self, rows, numRows=0, numCols=0):
+        """
+        Create a wrapper over a Java RowMatrix.
+
+        :param rows: An RDD of Vectors.
+        """
+        if not isinstance(rows, RDD):
+            raise TypeError("rows should be an RDD object, got %s" % type(rows))
+        first = rows.first()
+        if not isinstance(first, Vector):
+            raise TypeError("rows should be an RDD of Vectors, got an RDD of %s" % type(first))
+
+        javaRowMatrix = callMLlibFunc("createRowMatrix", rows, long(numRows), int(numCols))
+        self._jrm = JavaModelWrapper(javaRowMatrix)
+        self.rows = rows
+
+    @staticmethod
+    def _from_java(javaRowMatrix):
```

Comment:
Yeah I agree, it's not ideal, but it does keep the API clean.  An alternative would be to add an optional Java model parameter to each of the class constructors, and use it if present.  Then we could just construct the Java model on the Scala/Java side, grab it's rows, and pass both of those into the class constructors, bypassing the final step of reconstruction.  This parameter wouldn't be very useful for the end user though.

@mengxr Do you have any thoughts on this?